docs/check_sphinx.py

```diff
@@ -1,19 +1,30 @@
 import os
 import subprocess
+
 import pytest
+import pytest_dependency
 
-@pytest.fixture
-def build_docs(tmpdir):
+@pytest.fixture(scope="session")
+def build_docs(tmpdir_factory):
+    tmpdir = tmpdir_factory.mktemp("doctest")
     doctrees = tmpdir.join("doctrees")
     htmldir = tmpdir.join("html")
     docs_dir = os.path.dirname(os.path.abspath(__file__))
     # Ensure that all required files such as inmanta.pdf exist
-    subprocess.check_call(["make", "-C", docs_dir, "html", f"BUILDDIR={tmpdir}"])
-    return docs_dir, doctrees, htmldir
+    build_proc = subprocess.run(["make", "-C", docs_dir, "html", f"BUILDDIR={tmpdir}"], check=False)
+    return docs_dir, doctrees, htmldir, build_proc
+
+
+@pytest.mark.build
+@pytest.mark.dependency()
```

Comment:
why is this required?

Reply:
For pytest_dependency. Marking it like so makes the result of the test internally available so other tests can depend on it.

Reply:
I am not sure it is that important that on failure of the build the linkcheck is not executed. This means we would get fail,fail instead of fail,skip. That is not enough reason for me to add an additional dep to the requirements.

Reply:
Ok I will take it out. It doesn't make a big difference. The link-check doesn't take that long anyway.